From vapier@gentoo.org  Mon Jul 27 14:14:14 2009
From: Mike Frysinger <vapier@gentoo.org>
Date: Mon, 27 Jul 2009 14:20:56 -0400
Subject: driver core: simplify platform_get_resource()
To: Greg Kroah-Hartman <gregkh@suse.de>
Cc: linux-kernel@vger.kernel.org
Message-ID: <1248718856-22287-1-git-send-email-vapier@gentoo.org>


The platform_get_resource() function takes a number for the desired index
into the resource array, but then does a for loop on the array to get to
that index.  Considering the array is linear, the loop overhead is just
that -- overhead.  So unless I missed something, convert it into an index
check and access the desired resource directly.  Resulting code makes a
lot more sense considering its purpose.

Signed-off-by: Mike Frysinger <vapier@gentoo.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/platform.c |   10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

--- a/drivers/base/platform.c
+++ b/drivers/base/platform.c
@@ -37,14 +37,12 @@ EXPORT_SYMBOL_GPL(platform_bus);
 struct resource *platform_get_resource(struct platform_device *dev,
 				       unsigned int type, unsigned int num)
 {
-	int i;
-
-	for (i = 0; i < dev->num_resources; i++) {
-		struct resource *r = &dev->resource[i];
-
-		if (type == resource_type(r) && num-- == 0)
+	if (num >= 0 && num < dev->num_resources) {
+		struct resource *r = &dev->resource[num];
+		if (type == resource_type(r))
 			return r;
 	}
+
 	return NULL;
 }
 EXPORT_SYMBOL_GPL(platform_get_resource);
